relay/pkg/block/pull.go

```diff
@@ -0,0 +1,77 @@
+package block
+
+import (
+	"fmt"
+
+	"github.com/keep-network/tbtc/relay/pkg/btc"
+)
+
+func (f *Forwarder) pullNextHeader() error {
+	nextHeader, err := f.pullHeaderFromBtcChain(f.nextPullHeaderHeight)
+	if err != nil {
+		return fmt.Errorf(
+			"could not get header by height at [%d]: [%v]",
+			f.nextPullHeaderHeight,
+			err,
+		)
+	}
+
+	// TODO: Consider just comparing hashes - should be enough
+	// and check if nextHeader and lastPulledHeader header can ever be not equal
+	if !nextHeader.Equals(f.lastPulledHeader) {
+		f.pushHeaderToQueue(nextHeader)
+		f.lastPulledHeader = nextHeader
+		f.nextPullHeaderHeight++
+	}
+
+	return nil
+}
+
+func (f *Forwarder) pullHeaderFromBtcChain(height int64) (*btc.Header, error) {
+	return f.btcChain.GetHeaderByHeight(height)
+}
+
+func (f *Forwarder) pushHeaderToQueue(header *btc.Header) {
+	f.headersQueue <- header
+}
+
+func (f *Forwarder) setNextPullHeaderHeight(height int64) {
```

Comment:
There is no sense to make a setter for `nextPullHeaderHeight` in that case. We're not doing any validation or complex logic here. We can set the `nextPullHeaderHeight` field directly from calling code.